src/libponyc/codegen/codegen.c

```diff
@@ -882,7 +882,7 @@ bool codegen_init(pass_opt_t* opt)
   if(opt->features != NULL)
     opt->features = LLVMCreateMessage(opt->features);
   else
-    opt->features = LLVMCreateMessage("");
+    opt->features = LLVMGetHostCPUFeatures();
```

Comment:
@jemc The following diff works in my limited testing on linux. I had to change how the assertion was being done or else the CPU features weren't properly detected on a `config=release` build (according to my temporary `printf` validations).

```C
diff --git a/src/libponyc/codegen/codegen.c b/src/libponyc/codegen/codegen.c
index bbadb25..f39239e 100644
--- a/src/libponyc/codegen/codegen.c
+++ b/src/libponyc/codegen/codegen.c
@@ -882,12 +882,24 @@ bool codegen_init(pass_opt_t* opt)
   if(opt->features != NULL)
     opt->features = LLVMCreateMessage(opt->features);
   else
-    if((opt->cpu == LLVMGetHostCPUName()) && \
-      (opt->triple == default_triple_ifdef))
+  {
+    char* host_cpu = LLVMGetHostCPUName();
+#ifdef PLATFORM_IS_MACOSX
+    // This is to prevent XCode 7+ from claiming OSX 14.5 is required.
+    char* default_triple = LLVMCreateMessage("x86_64-apple-macosx");
+#else
+    char* default_triple = LLVMGetDefaultTargetTriple();
+#endif
+    if((strcmp(opt->cpu, host_cpu) == 0) && \
+      (strcmp(opt->triple, default_triple) == 0))
       opt->features = LLVMGetHostCPUFeatures();
     else
       opt->features = LLVMCreateMessage("");
 
+    free(host_cpu);
+    free(default_triple);
+  }
+
   return true;
 }
 
diff --git a/src/libponyc/codegen/host.cc b/src/libponyc/codegen/host.cc
index bea4196..417a4a4 100644
--- a/src/libponyc/codegen/host.cc
+++ b/src/libponyc/codegen/host.cc
@@ -36,7 +36,10 @@ char* LLVMGetHostCPUName()
 char* LLVMGetHostCPUFeatures()
 {
   StringMap<bool> features;
-  assert(sys::getHostCPUFeatures(features));
+  bool got_features = sys::getHostCPUFeatures(features);
+
+  assert(got_features);
+  (void)got_features;
 
   // Calculate the size of buffer that will be needed to return all features.
   size_t buf_size = 0;
```